boltons/classutils.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""Python's built-in :mod:`functools` module builds several useful
+utilities on top of Python's first-class function support.
+``classutils`` attempts to do the same for metaprogramming with
+classes, types and instances.
+"""
+
+import weakref
+
+
+def gencls(*bases, **extras):
+    """Define a new class inline (ie. class version of "lambda").
+    Takes classes to use as base class from positional arguments,
+    and any extra attributes of the class to set from keyword arguments.
+
+    Useful for once-off classes that can be described as "other class, but with a mixin"
+
+    >>> class A(object):
+    ...     name = 'Alice'
+    ...     def greet(self):
+    ...         print('Hello, ' + self.name)
+    ...
+    >>> class BobNameMixin(object):
+    ...     name = 'Bob'
+    ...
+    >>> gencls(BobNameMixin, A)().greet()
+    Hello, Bob
+    >>> gencls(A, name='Eve')().greet()
+    Hello, Eve
+
+    """
+    return type('gencls:{0}'.format(','.join(base.__name__ for base in bases)), bases, extras)
+
+
+def issubclass(subcls, supercls):
+    """As per the builtin issubclass(), but will simply return False if
+    either arg is not suitable (eg. if subcls is not a class),
+    instead of raising TypeError.
+    """
+    try:
+        return __builtins__['issubclass'](subcls, supercls)
+    except TypeError:
+        return False
+
+
+def get_all_subclasses(cls):
```

Comment:
It already is recursive. See line: `subs_of_subs = [get_all_subclasses(subcls) for subcls in subs]`